src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +252,121 @@ pub fn register(registry: &mut FunctionRegistry) {
                     .any(|(k, _)| k == key)
             },
         );
+
+    registry.register_3_arg_core::<EmptyMapType, GenericType<0>, GenericType<1>, MapType<GenericType<0>, GenericType<1>>, _, _>(
+        "map_insert",
+        |_, _, insert_key_domain, insert_value_domain| {
+            FunctionDomain::Domain(Some((
+                insert_key_domain.clone(),
+                insert_value_domain.clone(),
+            )))
+        },
+        |_, key, value, ctx| {
+            let mut b = ArrayType::create_builder(1, ctx.generics);
+            b.put_item((key.into_scalar().unwrap(), value.into_scalar().unwrap()));
+            b.commit_row();
+            return Value::Scalar(MapType::build_scalar(b));
+        },
+    );
+
+    registry.register_passthrough_nullable_3_arg(
```

Comment:
Use `register_2_arg_core` instead of `register_passthrough_nullable_3_arg`, because `register_passthrough_nullable_3_arg` will cause incorrect return values when the value is `NULL`, for example: `select map_insert({'a':'b'}, 'c', null)` should return `{'a':'b','c':NULL}` instead of `NULL`.
the function can be written as:
```rust
    registry.register_3_arg_core::<NullableType<MapType<GenericType<0>, GenericType<1>>>, GenericType<0>, GenericType<1>, _, _>
```